chat-plugins/blackjack.js

```diff
@@ -0,0 +1,397 @@
+				this.add(`<br /><b>The Dealer</b> stands with ${cards} (${this.dealer.points})`);
+				this.end();
+			}
+			return;
+		}
+		this.players[this.curUser].user.sendTo(this.room.id, '|uhtmlchange|user-bj-' + this.id + '|');
+		if (this.players[this.curUser].status !== 'playing') {
+			let number = 0;
+			while (this.players[Object.keys(this.players)[number]].status !== 'playing') number++;
+			this.curUser = Object.keys(this.players)[number];
+		}
+		let output = `|uhtml|user-bj-${this.id}|<div class="infobox">`;
+		output += "It's your turn to move<br />";
+		for (let u in this.players[this.curUser].cards) {
+			let card = this.players[this.curUser].cards[u];
+			output += `<img src="${IMAGES}${toId(card).toUpperCase() + this.symbols[card.substr(-1)]}.png" title="${card}" height="100"/>' `;
+		}
+		output += `<br />Score: ${this.players[this.curUser].points}`;
+		output += '<br /><button class="button" name="send" value="/blackjack hit" title="Hit (get another card)">Hit</button> | <button class="button" name="send" value="/blackjack stand" title="Stand (just keep these cards)">Stand</button>';
+
+		//this.players[this.curUser].user.sendTo(this.room.id, "|c:|" + (~~(Date.now() / 1000)) + "|~|" + this.players[this.curUser].user.name + ", it's your turn to play Blackjack.");
+		this.players[this.curUser].user.sendTo(this.room.id, output);
+		this.timer = setTimeout(() => {
+			clearTimeout(this.timer);
+			delete this.timer;
+			let cards = "";
+			for (let u in this.players[this.curUser].cards) cards += `[${this.players[this.curUser].cards[u]}] `;
+			this.players[this.curUser].status = 'stand';
+			this.add(Chat.html`<br /><b>${this.curUser}</b> stands with ${cards} (${this.players[this.curUser].points}) (Auto-stand: took too long to move)`);
+			this.next();
+		}, this.turnTimeoutMinutes * 60 * 1000);
+	}
+	joinGame(user) {
+		if (!user.named) return this.errorMessage(user, `You must be signed in to play Blackjack.`);
+		if (this.started) return this.errorMessage(user, `Blackjack has already started.`);
+		let joined = this.addPlayer(user, this.players);
+		if (!joined) {
+			this.errorMessage(user, `You are already in this game.`);
+			return false;
+		}
+		this.add(Chat.html`<br />${user.name} has joined blackjack.`);
+
+		if (Object.keys(this.players).length === this.playerCap) {
+			this.start();
+		}
+		return true;
+	}
+	leaveGame(user) {
+		if (this.started) return this.errorMessage(user, `You cannot leave this game; it has already started.`);
+		let left = this.removePlayer(user);
+		if (!left) return this.errorMessage(user, `You are not in this game to leave it.`);
+
+		this.add(Chat.html`<br />${user.name} has left blackjack.`);
+	}
+	errorMessage(user, message) {
+		user.sendTo(this.room, Chat.html`|html|<div class="error-message">${message}</div>`);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+		this.user = user;
```

Comment:
I just started using ``this.players[userid].sendRoom`` instead of holding onto the user object here